test/common/network/utility_test.cc

```diff
@@ -204,6 +204,32 @@ TEST(NetworkUtility, LocalConnection) {
 
   remote_addr.reset(new Network::Address::Ipv6Instance("fd00::"));
   EXPECT_FALSE(Utility::isLocalConnection(socket));
+
+  local_addr.reset(new Network::Address::Ipv4Instance("4.4.4.4"));
+  remote_addr.reset(new Network::Address::Ipv4Instance("4.4.4.4"));
+  EXPECT_TRUE(Utility::isLocalConnection(socket));
+
+  local_addr.reset(new Network::Address::Ipv6Instance("fabc::42"));
+  remote_addr.reset(new Network::Address::Ipv6Instance("fabc::42"));
+  EXPECT_TRUE(Utility::isLocalConnection(socket));
+
+  local_addr.reset(new Network::Address::Ipv4Instance("4.4.4.4", 80));
+  remote_addr.reset(new Network::Address::Ipv4Instance("4.4.4.4", 23413));
+  EXPECT_TRUE(Utility::isLocalConnection(socket));
+
+  local_addr.reset(new Network::Address::Ipv6Instance("fabc::42", 80));
+  remote_addr.reset(new Network::Address::Ipv6Instance("fabc::42", 23413));
+  EXPECT_TRUE(Utility::isLocalConnection(socket));
+
+  local_addr.reset(new Network::Address::Ipv4Instance("4.4.4.4", 80));
+  remote_addr.reset(new Network::Address::Ipv4Instance(
+      Utility::getLocalAddress(Address::IpVersion::v4)->ip()->addressAsString(), 23413));
+  EXPECT_TRUE(Utility::isLocalConnection(socket));
+
+  local_addr.reset(new Network::Address::Ipv6Instance("fabc::42", 80));
+  remote_addr.reset(new Network::Address::Ipv6Instance(
+      Utility::getLocalAddress(Address::IpVersion::v6)->ip()->addressAsString(), 23413));
+  EXPECT_TRUE(Utility::isLocalConnection(socket));
```

Comment:
Could you add a test comparing IPv4 and IPv6 addresses? It looks that we're missing it.

Reply:
Added.